hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieInputFormatUtils.java

```diff
@@ -481,9 +481,14 @@ public static Map<HoodieTableMetaClient, List<Path>> groupSnapshotPathsByMetaCli
     Map<HoodieTableMetaClient, List<Path>> grouped = new HashMap<>();
     metaClientList.forEach(metaClient -> grouped.put(metaClient, new ArrayList<>()));
     for (Path path : snapshotPaths) {
+      String inputPathStr = path.toString();
       // Find meta client associated with the input path
-      metaClientList.stream().filter(metaClient -> path.toString().contains(metaClient.getBasePath().toString()))
-          .forEach(metaClient -> grouped.get(metaClient).add(path));
+      Option<HoodieTableMetaClient> matchedMetaClient = Option.fromJavaOptional(metaClientList.stream()
+          .filter(metaClient -> {
+            String basePathStr = metaClient.getBasePath().toString();
+            return inputPathStr.equals(basePathStr) || inputPathStr.startsWith(basePathStr + "/"); })
```

Comment:
Shouldn't we do the same fix for checks in groupFileStatusForSnapshotPaths (461 and 463 lines)?

Reply:
The `groupFileStatusForSnapshotPaths` is not used anywhere. In the [HUDI-3094](https://github.com/apache/hudi/pull/4417) the `HoodieParquetInputFormat.listStatus()` which was the caller of this method was removed and related logic was moved to `InputPathHandler` where it belongs now. So I deleted this method entirely.